Packages that panic/fail before running any test are not reported

If a package has tests but fails in an `init()`, the package in the report has no indication of this failure.

In case of a panic in an init(), the output will look like:
```
panic: the description of the panic here

goroutine 1 [running]:
panic(....)
   /usr/local/go/src/runtime/panic.go:500....
...more stack frames...
...more goroutines...
FAIL  package/name 0.01s
```

--- a/go-junit-report_test.go
+++ b/go-junit-report_test.go
@@ -561,6 +561,55 @@ var testCases = []TestCase{
 			},
 		},
 	},
+	{
+		name:       "14-panic.txt",
+		reportName: "14-report.xml",
+		report: &parser.Report{
+			Packages: []parser.Package{
+				{
+					Name: "package/panic",
+					Time: 3,
+					Tests: []*parser.Test{
+						{
+							Name:   "Failure",
+							Result: parser.FAIL,
+							Output: []string{
+								"panic: init",
+								"stacktrace",
+							},
+						},
+					},
+				},
+				{
+					Name: "package/panic2",
+					Time: 3,
+					Tests: []*parser.Test{
+						{
+							Name:   "Failure",
+							Result: parser.FAIL,
+							Output: []string{
+								"panic: init",
+								"stacktrace",
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		name:       "15-empty.txt",
+		reportName: "15-report.xml",
+		report: &parser.Report{
+			Packages: []parser.Package{
+				{
+					Name:  "package/empty",
+					Time:  1,
+					Tests: []*parser.Test{},
+				},
+			},
+		},
+	},
 }
 
 func TestParser(t *testing.T) {

--- a/parser/parser.go
+++ b/parser/parser.go
@@ -72,6 +72,9 @@ func Parse(r io.Reader, pkgName string) (*Report, error) {
 	// the name of the package which it's build failure output is being captured
 	var capturedPackage string
 
+	// capture any non-test output
+	var buffer []string
+
 	// parse lines
 	for {
 		l, _, err := reader.ReadLine()
@@ -106,6 +109,15 @@ func Parse(r io.Reader, pkgName string) (*Report, error) {
 					Result: FAIL,
 					Output: packageCaptures[matches[2]],
 				})
+			} else if matches[1] == "FAIL" && len(tests) == 0 && len(buffer) > 0 {
+				// This package didn't have any tests, but it failed with some
+				// output. Create a dummy test with the output.
+				tests = append(tests, &Test{
+					Name:   "Failure",
+					Result: FAIL,
+					Output: buffer,
+				})
+				buffer = buffer[0:0]
 			}
 
 			// all tests in this package are finished
@@ -116,6 +128,7 @@ func Parse(r io.Reader, pkgName string) (*Report, error) {
 				CoveragePct: coveragePct,
 			})
 
+			buffer = buffer[0:0]
 			tests = make([]*Test, 0)
 			coveragePct = ""
 			cur = ""
@@ -157,6 +170,9 @@ func Parse(r io.Reader, pkgName string) (*Report, error) {
 		} else if capturedPackage != "" {
 			// current line is build failure capture for the current built package
 			packageCaptures[capturedPackage] = append(packageCaptures[capturedPackage], line)
+		} else {
+			// buffer anything else that we didn't recognize
+			buffer = append(buffer, line)
 		}
 	}
 

--- a/tests/14-panic.txt
+++ b/tests/14-panic.txt
@@ -1,0 +1,6 @@
+panic: init
+stacktrace
+FAIL    package/panic  0.003s
+panic: init
+stacktrace
+FAIL    package/panic2  0.003s

--- a/tests/14-report.xml
+++ b/tests/14-report.xml
@@ -1,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites>
+	<testsuite tests="1" failures="1" time="0.003" name="package/panic">
+		<properties>
+			<property name="go.version" value="1.0"></property>
+		</properties>
+		<testcase classname="panic" name="Failure" time="0.000">
+			<failure message="Failed" type="">panic: init&#xA;stacktrace</failure>
+		</testcase>
+	</testsuite>
+	<testsuite tests="1" failures="1" time="0.003" name="package/panic2">
+		<properties>
+			<property name="go.version" value="1.0"></property>
+		</properties>
+		<testcase classname="panic2" name="Failure" time="0.000">
+			<failure message="Failed" type="">panic: init&#xA;stacktrace</failure>
+		</testcase>
+	</testsuite>
+</testsuites>

--- a/tests/15-empty.txt
+++ b/tests/15-empty.txt
@@ -1,0 +1,3 @@
+testing: warning: no tests to run
+PASS
+ok  	package/empty	0.001s

--- a/tests/15-report.xml
+++ b/tests/15-report.xml
@@ -1,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites>
+	<testsuite tests="0" failures="0" time="0.001" name="package/empty">
+		<properties>
+			<property name="go.version" value="1.0"></property>
+		</properties>
+	</testsuite>
+</testsuites>

